apps/src/p5lab/spritelab/CoreLibrary.js

```diff
@@ -107,6 +111,39 @@ export default class CoreLibrary {
     });
   }
 
+  drawVariableBubbles() {
+    this.variableBubbles.forEach(variable => {
+      const {name, location} = variable;
+      if (!name.length || !location) {
+        return;
+      }
+
+      const value = this.getVariableValue(name);
+      const text = `${name}: ${value}`;
+
+      // TODO: Confirm this handles shadow block locations appropriately
+      drawUtils.variableBubble(this.p5, location.x, location.y, text);
+    });
+  }
+
+  getVariableValue(variableName) {
+    if (!this.jsInterpreter) {
+      console.error('JS Interpreter not set in CoreLibrary');
+      return;
+    }
+
+    try {
+      // TODO: Check for existence of this.REFERENCE_ERROR and others in result,
+      // Or move this logic into the JSInterpreter and guarantee that it always returns a string
+      // TODO: Decide on what to display for undefined variables
+      const result = this.jsInterpreter.evaluateWatchExpression(variableName);
```

Comment:
for my information: what is the reason we need to use the js interpreter for this? Can you add a comment explaining why we need it?

Reply:
Added
```
// Blockly does not execute code or track the runtime values of variables, so we need to
// evaluate the variable's value using the JSInterpreter.
```

Does that cover it?